src/InteractiveWindow/Editor/InteractiveWindow.UIThreadOnly.cs

```diff
@@ -1378,6 +1403,11 @@ private void RemoveLastInputPrompt()
             private void AddLanguageBuffer()
             {
                 ITextBuffer buffer = _factory.CreateAndActivateBuffer(_window);
+                if (CurrentLanguageBuffer != null)
+                {
+                    _textBufferUndoManagerProvider.RemoveTextBufferUndoManager(CurrentLanguageBuffer);
+                }
+                UndoManager = _textBufferUndoManagerProvider.GetTextBufferUndoManager(buffer);
```

Comment:
Ah, looking at https://github.com/dotnet/roslyn/blob/master/src/Interactive/EditorFeatures/Core/Implementation/Interactive/InteractiveTextUndoHistoryWorkspaceServiceFactory.cs it looks like that is what you intend to do. I recall having a conversation long ago with @KevinH-MS (I think? :smile:) about this and we weren't sure where it landed. I'm still worried about that call to RemoveTextBufferUndoManager -- was that not happening anywhere previously? That is...bad,